src/main/java/com/guciowons/shoppingbasket/Basket/BasketService.java

```diff
@@ -0,0 +1,45 @@
+package com.guciowons.shoppingbasket.Basket;
+
+import com.guciowons.shoppingbasket.Product.Product;
+import com.guciowons.shoppingbasket.Product.ProductDao;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BasketService {
+    private Basket basket = new Basket();
+
+    private final ProductDao productDao;
+
+    public BasketService(ProductDao productDao) {
+        this.productDao = productDao;
+    }
+
+    public String addProductToBasket(int id, int quantity) {
+        Product product = productDao.findById(id).orElse(null);
+        if(product != null){
+            basket.addProduct(product, quantity);
+            return "Done";
+        }
+        return "No product";
```

Comment:
Błąd w zapisie produktu powinien być pokazany przez odpowiedni status w RESTowym API.

Któryś z kodów 4xx będzie okej. 

https://en.wikipedia.org/wiki/List_of_HTTP_status_codes

To wszystko powinno sie odbywać w warstwie RESTa, a nie w serwisie, więc tutaj powinieneś po prostu rzucić jakiś wyjątek, który powie, że coś się nie udało.